infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/secure/jwtproxy/JwtProxyConfigBuilder.java

```diff
@@ -119,6 +147,10 @@ public String build() throws InternalInfrastructureException {
         verifierConfig.setAuthUrl(authPageUrl.toString());
```

Comment:
Do we really need to play with `authErrorRedirectUriPrefix`? Would not it be simpler if Che Server only know about such prefix and provide jwt-proxy ready to use `authPageUrl` with prefix included?

Reply:
Well, the prefix is need for situations where there is an unauthed request which fails jwt verification. Then the flow is basically - go to the auth page and then go back to whereever the original request came from.

There lies the problem, because we're still using a target-rewriting ingress - so by the time the request reaches the JWT proxy, we'll have lost the full path of the original request, because the ingress rewrote it. Which means that JWT proxy cannot correctly construct the "whereever the original request came from" URL.  The `authErrorRedirectUriPrefix` is basically a way of restoring that lost piece of the URL.

Alternatively, I think it could work if we picked and chose carefully what ingresses where target-rewriting and what would not, but I didn't go down that route because I don't know the full consequence of that...

Reply:
Looks confusing or I still misunderstand something.
From naming I expect that jwtproxy will redirect to `authErrorRedirectUriPrefix` + `authErrorRedirectUri`, but what I got now: 
jwtProxy redirect to `authErrorRedirectUri + "?redirectUri=" + authErrorRedirectUriPrefix`, is it correct?
If yes - then maybe it would be better to name `authErrorRedirectUriPrefix` as `basePath` or something like that that would show the root path of jwtproxy application behind it.

Reply:
The redirect URI is constructed on the JWT side upon receiving the original unauthed request like this:
```go
protocol + "://" + req.Host + authErrorRedirectUriPrefix + req.URL.String()
```
where `req` is the original request. I wanted the name to highlight how this is actually used but `basePath` would work for me, too, even though not ideal either. It still doesn't give you an idea for why this is needed at all :) Let me think about a better name some more.

Reply:
Renamed to `public_base_path`.